src/app/shared/components/product-item/product-item.component.spec.ts

```diff
@@ -25,4 +62,71 @@ describe('ProductItemComponent', () => {
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('resetDefaultThumbnail should set preview image', () => {
+    component.product = product;
+    component.resetDefaultThumbnail();
+    expect(component.currentThumbnail).toBe(
+      `url(${product.thumbnailImageSrc})`
+    );
+  });
+
+  it('should return undefined if we call stop', () => {
+    fixture = TestBed.createComponent(ProductItemComponent);
+    component = fixture.componentInstance;
+    component.product = product;
+    const functionsReturn = component.hoverStateIn();
+    expect(functionsReturn).toBeUndefined();
+  });
+
+  it('should return undefined if we call stop', () => {
+    fixture = TestBed.createComponent(ProductItemComponent);
+    component = fixture.componentInstance;
+    component.product = product;
+    const functionsReturn = component.hoverStateOut();
+    expect(functionsReturn).toBeUndefined();
+  });
+
+  it('should return undefined if we call stop', () => {
+    fixture = TestBed.createComponent(ProductItemComponent);
+    component = fixture.componentInstance;
+    component.product = product;
+    const functionsReturn = component.onMouseLeave();
+    expect(functionsReturn).toBeUndefined();
```

Comment:
this test doesn't make sense
check the onMouseLeave is really called on mouse-leave event